File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n=numbers.length;
        for(int i=0; i<n; i++){
            int l=i+1, r=n-1;
            while(l<r){
                int mid = l+ (r-l)/2;
                if(numbers[mid] < target-numbers[i]){
                    l=mid+1;
                }
                else{
                    r=mid;
                }
            }
            if(numbers[i]+numbers[l] == target){
                return new int[]{i+1, l+1};
            }
        }
        return new int[]{0,0};
    }
}

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(numbers)):
            if target-numbers[i] in map:
                break
            map[numbers[i]]=i
        return [map[target-numbers[i]]+1, i+1]